pkg/environment/environment_provider.go

```diff
@@ -0,0 +1,55 @@
+package environment
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "k8s.io/client-go/kubernetes/typed/core/v1"
+)
+
+const defaultNamespace = "keptn"
+const secretPrefix = "locust"
+
+type EnvironmentProvider struct {
+	KubeAPI                v1.CoreV1Interface
+	KeptnNamespaceProvider StringSupplier
+}
+
+func NewEnvironmentProvider(kubeAPI v1.CoreV1Interface) *EnvironmentProvider {
+	return &EnvironmentProvider{
+		KubeAPI:                kubeAPI,
+		KeptnNamespaceProvider: envBasedStringSupplier("POD_NAMESPACE", defaultNamespace),
+	}
+}
+
+func (e EnvironmentProvider) PrepareEnvironment(project string, stage string, service string) []string {
```

Comment:
Please add a comment here, e.g.,
```
// PrepareEnvironment creates a list of environment variables based on project, stage and service
```

Reply:
I refactored the tests to make sure the generated environment is correct in the context of a command execution. Also added a test for more complex environment variables.